### PR TITLE
nanorange: update for conan v2 + replace with a new version

### PR DESCRIPTION
**Pull request by prince-chrismc**
Specify library name and version:  **lib/1.0**

Complete refresh for https://github.com/conan-io/conan-center-index/pull/16367#issuecomment-1491024681

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/nanorange/all/conanfile.py [hunk 1]
```diff
@@ -11,50 +14,52 @@ class NanorangeConan(ConanFile):
...
+            )
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def package(self):
+        copy(self, pattern="LICENSE_1_0.txt", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        copy(
+            self,
+            pattern="*.h",
```
Comment (RubenRBS) [inline recipes/nanorange/all/conanfile.py:57]: (Forgot to send message in the old review). NanoRange uses `.hpp`. Does this also copy those?

Comment (prince-chrismc) [reply recipes/nanorange/all/conanfile.py:57]: Nop! Good catch 👏  just cipy paste error by me :)

```suggestion
            pattern="*.hpp",
```

**Comment by prince-chrismc** [comment]
> Nice! I like this PR as a reminder of how much Conan 2 improves recipes

It's been 3 years since this was touched. There's a lot of 1.x Improvements too :)

**Review by RubenRBS — DISMISSED** [review]

File: recipes/nanorange/all/conanfile.py [hunk 2]
```diff
@@ -1,7 +1,10 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy
```
Comment (SpaceIm) [inline recipes/nanorange/all/conanfile.py:4]:
```suggestion
from conan.tools.files copy, get
```

File: recipes/nanorange/all/conanfile.py [hunk 3]
```diff
@@ -1,7 +1,10 @@
...
+from conan.tools.layout import basic_layout
+from conan.tools.scm import Version
 import os
```
Comment (SpaceIm) [inline recipes/nanorange/all/conanfile.py:7]:
```suggestion
import os

required_conan_version = ">=1.50.0"
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

**Review by RubenRBS — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Pull request merged**